spring-messaging/src/main/java/org/springframework/messaging/simp/stomp/StompCommand.java

```diff
@@ -32,46 +30,37 @@
 public enum StompCommand {
 
 	// client
-	CONNECT,
-	STOMP,
-	DISCONNECT,
-	SUBSCRIBE,
-	UNSUBSCRIBE,
-	SEND,
-	ACK,
-	NACK,
-	BEGIN,
-	COMMIT,
-	ABORT,
+	CONNECT(SimpMessageType.CONNECT),
+	STOMP(SimpMessageType.CONNECT),
+	DISCONNECT(SimpMessageType.DISCONNECT),
+	SUBSCRIBE(SimpMessageType.SUBSCRIBE),
+	UNSUBSCRIBE(SimpMessageType.UNSUBSCRIBE),
+	SEND(SimpMessageType.MESSAGE),
+	ACK(null),
+	NACK(null),
+	BEGIN(null),
+	COMMIT(null),
+	ABORT(null),
 
 	// server
-	CONNECTED,
-	MESSAGE,
-	RECEIPT,
-	ERROR;
-
-
-	private static Map<StompCommand, SimpMessageType> messageTypes = new HashMap<>();
-	static {
-		messageTypes.put(StompCommand.CONNECT, SimpMessageType.CONNECT);
-		messageTypes.put(StompCommand.STOMP, SimpMessageType.CONNECT);
-		messageTypes.put(StompCommand.SEND, SimpMessageType.MESSAGE);
-		messageTypes.put(StompCommand.MESSAGE, SimpMessageType.MESSAGE);
-		messageTypes.put(StompCommand.SUBSCRIBE, SimpMessageType.SUBSCRIBE);
-		messageTypes.put(StompCommand.UNSUBSCRIBE, SimpMessageType.UNSUBSCRIBE);
-		messageTypes.put(StompCommand.DISCONNECT, SimpMessageType.DISCONNECT);
-	}
+	CONNECTED(null),
+	MESSAGE(SimpMessageType.MESSAGE),
+	RECEIPT(null),
+	ERROR(null);
 
 	private static Collection<StompCommand> destinationRequired = Arrays.asList(SEND, SUBSCRIBE, MESSAGE);
 	private static Collection<StompCommand> subscriptionIdRequired = Arrays.asList(SUBSCRIBE, UNSUBSCRIBE, MESSAGE);
 	private static Collection<StompCommand> contentLengthRequired = Arrays.asList(SEND, MESSAGE, ERROR);
 	private static Collection<StompCommand> bodyAllowed = Arrays.asList(SEND, MESSAGE, ERROR);
 
+	private SimpMessageType simpMessageType;
 
+	StompCommand(SimpMessageType simpMessageType) {
+		this.simpMessageType = simpMessageType;
+	}
 
 	public SimpMessageType getMessageType() {
-		SimpMessageType type = messageTypes.get(this);
-		return (type != null) ? type : SimpMessageType.OTHER;
+		return (simpMessageType != null) ? simpMessageType : SimpMessageType.OTHER;
```

Comment:
Instead of checking for null here, why not just pass SimpMessageType.OTHER instead of null to the constructors?